src/rdf4cpp/rdf/datatypes/xsd/Decimal.hpp

```diff
@@ -49,10 +49,8 @@ template<>
 inline std::string LiteralDatatypeImpl<xsd_decimal>::to_string(const cpp_type &value) {
 
     std::ostringstream str_os;
-    // Set Fixed -Point Notation
-    str_os << std::fixed;
-    str_os << value;
-    // Get string from output string stream
+    // Set Fixed -Point Notation for Decimal
+    str_os << std::fixed << value;
```

Comment:
This does not produce a canonical representation. We need "2.0" not "2.000000". See: https://www.w3.org/TR/xmlschema-2/#decimal

You could add fmt 8.1.1 via [fetchcontent](https://cmake.org/cmake/help/latest/module/FetchContent.html) and link it as PRIVATE to rdf4cpp's target link libraries. (If it helps). If you use fmt, you should add an cpp file for Decimal.hpp and include the `<fmt/format.h>` only in that cpp file. The rational behind this is that we do not want to include all of fmt's headers with rdf4cpp. 

The canonization should produce the following results: 
- "0.000000031" -> "0.000000031"
- "1", "1.0" -> "1.0"
- "10000000" -> "10000000.0"

Those resources might also be relevant:
- https://stackoverflow.com/questions/1521607/check-double-variable-if-it-contains-an-integer-and-not-floating-point
- https://www.zverovich.net/2019/02/11/formatting-floating-point-numbers.html